tests/ImageSharp.Tests/Formats/Pbm/PbmEncoderTests.cs

```diff
@@ -0,0 +1,144 @@
+        {
+            var options = new PbmEncoder();
+
+            var testFile = TestFile.Create(imagePath);
+            using (Image<Rgba32> input = testFile.CreateRgba32Image())
+            {
+                using (var memStream = new MemoryStream())
+                {
+                    input.Save(memStream, options);
+                    memStream.Position = 0;
+                    using (var output = Image.Load<Rgba32>(memStream))
+                    {
+                        PbmMetadata meta = output.Metadata.GetPbmMetadata();
+                        Assert.Equal(pbmColorType, meta.ColorType);
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(PbmColorTypeFiles))]
+        public void PbmEncoder_WithPlainEncoding_PreserveBitsPerPixel(string imagePath, PbmColorType pbmColorType)
+        {
+            var options = new PbmEncoder()
+            {
+                Encoding = PbmEncoding.Plain
+            };
+
+            TestFile testFile = TestFile.Create(imagePath);
+            using (Image<Rgba32> input = testFile.CreateRgba32Image())
+            {
+                using (var memStream = new MemoryStream())
+                {
+                    input.Save(memStream, options);
+                    memStream.Position = 0;
+                    using (var output = Image.Load<Rgba32>(memStream))
+                    {
+                        PbmMetadata meta = output.Metadata.GetPbmMetadata();
+                        Assert.Equal(pbmColorType, meta.ColorType);
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [WithFile(BlackAndWhitePlain, PixelTypes.Rgb24)]
+        public void PbmEncoder_P1_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.BlackAndWhite, PbmEncoding.Plain);
+
+        [Theory]
+        [WithFile(BlackAndWhiteBinary, PixelTypes.Rgb24)]
+        public void PbmEncoder_P4_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.BlackAndWhite, PbmEncoding.Binary);
+
+        /* Disabled as Magick throws an error reading the input image
+        [Theory]
+        [WithFile(GrayscalePlain, PixelTypes.Rgb24)]
+        public void PbmEncoder_P2_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.Grayscale, PbmEncoding.Plain);
+        */
+
+        [Theory]
+        [WithFile(GrayscaleBinary, PixelTypes.Rgb24)]
+        public void PbmEncoder_P5_Works<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel> => TestPbmEncoderCore(provider, PbmColorType.Grayscale, PbmEncoding.Binary);
+
+        /* Disabled as Magick throws an error reading the input image
```

Comment:
I wonder why magick fails here, @ynse01 are you sure this image is valid? Maybe we should raise an issue in the magick repo then?

Reply:
I think there is a missing line feed

Reply:
I took these examples from the offcial specification website and the linux man page of libnetpbm.
The specification makes no distiction btween different types of whitespace. So, line feeds are not significant.
I consider these failures a bug in Magick. I'll post an issue about it on their repo.

Reply:
As above, while we wait for Magick to fix the bug, I placed some Magick compatible input images.